src/test/java/org/candlepin/subscriptions/db/model/HostTest.java

```diff
@@ -0,0 +1,109 @@
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.db.model;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.candlepin.subscriptions.inventory.db.model.InventoryHostFacts;
+import org.candlepin.subscriptions.tally.facts.NormalizedFacts;
+
+import org.junit.jupiter.api.Test;
+
+import java.time.OffsetDateTime;
+import java.util.UUID;
+
+
+class HostTest {
+
+    InventoryHostFacts inventoryHostFacts = new InventoryHostFacts();
+    NormalizedFacts normalizedFacts = new NormalizedFacts();
+
+    @Test
+    void populateFieldsFromHbiNull() {
+        final Host host = new Host();
+
+        host.populateFieldsFromHbi(inventoryHostFacts, normalizedFacts);
+
+        assertNull(host.getInventoryId());
+        assertNull(host.getInsightsId());
+        assertNull(host.getAccountNumber());
+        assertNull(host.getOrgId());
+        assertNull(host.getDisplayName());
+        assertNull(host.getSubscriptionManagerId());
+        assertFalse(host.getGuest());
+        assertNull(host.getHypervisorUuid());
+        assertEquals(0, host.getMeasurements().size());
+        assertFalse(host.isHypervisor());
+        assertNull(host.getCloudProvider());
+        assertNull(host.getLastSeen());
+        assertNull(host.getHardwareType());
+    }
+
+    @Test
+    void populateFieldsFromHbiFull() {
+        final Host host = new Host();
+
+        inventoryHostFacts = getInventoryHostFactsFull();
+        normalizedFacts = getNormalizedFactsFull();
+
+        host.populateFieldsFromHbi(inventoryHostFacts, normalizedFacts);
+
+        assertEquals(host.getInventoryId(), inventoryHostFacts.getInventoryId().toString());
+        assertEquals(host.getInsightsId(), inventoryHostFacts.getInsightsId());
+        assertEquals(host.getAccountNumber(), inventoryHostFacts.getAccount());
+        assertEquals(host.getOrgId(), inventoryHostFacts.getOrgId());
+        assertEquals(host.getDisplayName(), inventoryHostFacts.getDisplayName());
+        assertEquals(host.getSubscriptionManagerId(), inventoryHostFacts.getSubscriptionManagerId());
+        assertTrue(host.getGuest());
+        assertEquals(host.getHypervisorUuid(), normalizedFacts.getHypervisorUuid());
+        assertEquals(2, host.getMeasurements().size());
+        assertTrue(host.isHypervisor());
+        assertEquals(host.getCloudProvider(), normalizedFacts.getCloudProviderType().name());
+        assertEquals(host.getLastSeen(), inventoryHostFacts.getModifiedOn());
+        assertEquals(host.getHardwareType(), normalizedFacts.getHardwareType());
+    }
+
+    private InventoryHostFacts getInventoryHostFactsFull() {
+        inventoryHostFacts.setInventoryId(UUID.randomUUID());
```

Comment:
It's weird to have a "get" method with side effects - setting the attributes of a shared object - at which point you don't even need to return the object because it updated at the class level and can be accessed as such.

I would prefer to see one of the following options:

1.  Remove inventoryHostFacts and normalizedFacts as class level variables, and put the contents of these two methods local to the test they're being used for, since the methods only get invoked in one place.
2.  Remove inventoryHostFacts and normalizedFacts as class level variables, and make these helper methods return a new instance of InventoryHostFacts/NormalizedFacts
```java
private InventoryHostFacts getNewInventoryHostFactsFull() {
        InventoryHostFacts inventoryHostFacts = new InventoryHostFacts();
        inventoryHostFacts.setInventoryId(UUID.randomUUID());
        // .....
       return inventoryHostFacts;
```
3.  If you wanted all tests to share the same instance of these objects (doesn't seem like that's what we want to do), you could initialize them with a method annotated with `@Before`

Reply:
I'll take option 2 above, since it's easy and results in smaller blocks of code than option 1.  Option 3 doesn't work with the first test with all-null fields.  New PR forthcoming.